### PR TITLE
jobs/build: request as many CPUs as parallel VMs and run reprovision tests serially

### PR DESCRIPTION
**Pull request by jlebon**
```
commit c9fc772f7c17d77b94b6254cbab0888d63c7e736
Date:   Thu Aug 4 19:23:35 2022 -0400

    jobs/build: request as many CPUs as parallel VMs

    Right now we're running with `--parallel 5` and run the upgrade tests in
    parallel, but only have 4 CPUs. This causes some heavier tests to fail.
    Let's request as many CPUs as we run VM tests in parallel. Since the
    latter is calculated from the memory request, that's what we do.

    In effect, this bumps up our CPU requests to 6.
```
```
commit 04cb8c388688cfb4a11f1db1381aca2a37998df6
Author: Jonathan Lebon <jonathan@jlebon.com>
Date:   Fri Aug 5 11:21:12 2022 -0400

    jobs/build: run reprovision tests not in parallel

    Those tests take 4G of RAM each and are heavy in I/O. Run them serially
    to stay within our declared memory request and minimize the chances of
    timeouts.
```

### CODE REVIEW

**Comment by jlebon** [comment]
The background for this is that we're working on moving the RHCOS pipeline to a new cluster (and basing it on the FCOS pipeline), and the reprovisioning tests there are much flakier (slower/more contended disk I/O?).

I think eventually we should move our x86_64 build to also use `podman --remote` to be consistent with the other arches. We can then use a VM with actual IOPS guarantees, which Kubernetes can't offer us AFAIK.

**Comment by jlebon** [comment]
Requires: https://github.com/coreos/coreos-assembler/pull/3028

File: jobs/build.Jenkinsfile
Comment (dustymabe) [inline jobs/build.Jenkinsfile:95]: maybe let's add a better comment here on why we're basing CPU requests on memory (which seems odd to me). basically what we're doing here is saying our limiting factor is memory and our VMs are approximately 1G memory per VM so request that many CPUs.

A more direct way of doing this may be defining first here how "parallel" we want our kola tests to run and then deriving both CPU and memory from that.

i.e. if we want a parallel of 6 (5 kola + 1 kola upgrade) then we define parallel of 6 and we need 6*1G+512M of memory and 6 cpus.

Comment (jlebon) [reply jobs/build.Jenkinsfile:98]: Though technically currently we're defining the amount of memory we need based on xz, not based on kola parallelism. The latter is being derived from that. I've added a comment to expand on that.

I think we could flip it around and say kola tests is our most memory intensive factor though it's mostly playing with words unless we actually want to go past 6.

**Review by dustymabe — APPROVED** [review]
LGTM

**Comment by jlebon** [comment]
Let's wait until releases are done before merging this.

**Comment by jlebon** [comment]
We're past the critical point in the releases. Merging.

**Pull request merged**
